python SDK build error due to api doc changes and form handling

this looks likely to be due to changes in the API doc from Looker

`Agent` class:
```
looker_sdk/sdk/api40/models.py:114: error: Name "date" is not defined  [name-defined]
looker_sdk/sdk/api40/models.py:115: error: Name "date" is not defined  [name-defined]
looker_sdk/sdk/api40/models.py:137: error: Name "date" is not defined  [name-defined]
looker_sdk/sdk/api40/models.py:138: error: Name "date" is not defined  [name-defined]
```

`ChatMessage` class:
```
looker_sdk/sdk/api40/models.py:1453: error: Name "date" is not defined  [name-defined]
looker_sdk/sdk/api40/models.py:1461: error: Name "date" is not defined  [name-defined]
```

`Conversation` class:
```
looker_sdk/sdk/api40/models.py:3570: error: Name "date" is not defined  [name-defined]
looker_sdk/sdk/api40/models.py:3571: error: Name "date" is not defined  [name-defined]
looker_sdk/sdk/api40/models.py:3588: error: Name "date" is not defined  [name-defined]
looker_sdk/sdk/api40/models.py:3589: error: Name "date" is not defined  [name-defined]
```

---

and the following part is likely due to changes to fix form handling for #1646 
```
looker_sdk/sdk/api40/methods.py:397: error: Dict entry 0 has incompatible type "str": "Optional[str]"; expected "str": "str"  [dict-item]
looker_sdk/sdk/api40/methods.py:397: error: Dict entry 1 has incompatible type "str": "Optional[str]"; expected "str": "str"  [dict-item]
```

--- a/packages/sdk-codegen/src/python.gen.ts
+++ b/packages/sdk-codegen/src/python.gen.ts
@@ -523,6 +523,7 @@ ${this.hooks.join('\n')}
       any: { default: this.nullStr, name: 'Any' },
       boolean: { default: this.nullStr, name: 'bool' },
       byte: { default: this.nullStr, name: 'bytes' },
+      date: { default: this.nullStr, name: 'datetime.datetime' },
       datetime: { default: this.nullStr, name: 'datetime.datetime' },
       double: { default: this.nullStr, name: 'float' },
       float: { default: this.nullStr, name: 'float' },

